frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -425,52 +406,34 @@ end
 
 function ReaderBookmark:gotoBookmark(pn_or_xp, marker_xp)
     if pn_or_xp then
-        local event = self.ui.document.info.has_pages and "GotoPage" or "GotoXPointer"
+        local event = self.ui.paging and "GotoPage" or "GotoXPointer"
         self.ui:handleEvent(Event:new(event, pn_or_xp, marker_xp))
     end
 end
 
--- This function adds "chapter" property to highlights already saved in the document
-function ReaderBookmark:updateHighlightsIfNeeded()
-    local version = self.ui.doc_settings:readSetting("bookmarks_version") or 0
-    if version >= 20200615 then
-        return
-    end
-
-    for page, highlights in pairs(self.view.highlight.saved) do
-        for _, highlight in pairs(highlights) do
-            local pg_or_xp = self.ui.document.info.has_pages and
-                    page or highlight.pos0
-            local chapter_name = self.ui.toc:getTocTitleByPage(pg_or_xp)
-            highlight.chapter = chapter_name
-        end
-    end
-
-    for _, bookmark in ipairs(self.bookmarks) do
-        if bookmark.pos0 then
-            local pg_or_xp = self.ui.document.info.has_pages and
-                    bookmark.page or bookmark.pos0
-                local chapter_name = self.ui.toc:getTocTitleByPage(pg_or_xp)
-            bookmark.chapter = chapter_name
-        elseif bookmark.page then -- dogear bookmark
-            local chapter_name = self.ui.toc:getTocTitleByPage(bookmark.page)
-            bookmark.chapter = chapter_name
-        end
-    end
-    self.ui.doc_settings:saveSetting("bookmarks_version", 20200615)
-end
-
 function ReaderBookmark:onShowBookmark(match_table)
     self.select_mode = false
     self.filtered_mode = match_table and true or false
-    self:updateHighlightsIfNeeded()
     -- build up item_table
     local item_table = {}
     local is_reverse_sorting = G_reader_settings:nilOrTrue("bookmarks_items_reverse_sorting")
-    local curr_page = self.ui.rolling and self.ui.document:getXPointer() or self.ui.paging.current_page
-    curr_page = self:getBookmarkPageString(curr_page)
+    local curr_page_num = self:getCurrentPageNumber()
+    local curr_page_string = self:getBookmarkPageString(curr_page_num)
+    local curr_page_item = {
+        page = curr_page_num,
+        text = DISPLAY_PREFIX["curr_page"] .. _("Book current page"),
```

Comment:
Just current page should do the trick.
```suggestion
        text = DISPLAY_PREFIX["curr_page"] .. _("Current page"),
```

Reply:
We have a button to jump to the bookmark list page containing the book curent page.

<img width="250" alt="04" src="https://user-images.githubusercontent.com/62179190/205588281-b6874ff4-fae9-45cb-b3de-3a49cdd5b9d4.png">

Reply:
Is there a particular reason that one doesn't just say "current page"? :-)

Reply:
Oh, yeah, that's extremely clunky wording (in both cases ;p).

Reply:
i.e., I'd go with something like "On the current page" for the popup, and a plain "Current page" for the new stuff.